course1/module7/assignment2/Question4.js

```diff
@@ -0,0 +1,53 @@
+// Given 2 arrays with related objects, return a new array where objects having the same id from each of the arrays are merged. Try to achieve it with a complexity - O(n).
+
+let userNames = [
+  {
+    id: 1,
+    first_name: "Nicki",
+  },
+  {
+    id: 2,
+    first_name: "Raychel",
+  },
+  {
+    id: 3,
+    first_name: "Demetris",
+  },
+  {
+    id: 4,
+    first_name: "Amata",
+  },
+];
+let userEmails = [
+  {
+    id: 2,
+    email: "rmcgrady1@cpanel.net",
+  },
+  {
+    id: 1,
+    email: "ncrozier0@squarespace.com",
+  },
+  {
+    id: 4,
+    email: "abraiden3@canalblog.com",
+  },
+  {
+    id: 3,
+    email: "dkilshall2@elpais.com",
+  },
+];
+
+console.log(mergeById(userNames, userEmails));
+
+function mergeById(userNames, userEmails) {
+  let mergedArray = new Array();
+  userNames.forEach((element) => {
+    userEmails.find(function (post, index) {
+      if (userEmails[index].id == element.id) {
```

Comment:
You have missed a case here. What if for an object in userNames we don't have object with similar Id in userEmails. In that case we won't have any object in final array which is not right. We should have the userNames object in final array